arctic/store/_pickle_store.py

```diff
@@ -54,7 +55,19 @@ def read(self, mongoose_lib, version, symbol, **kwargs):
                     data = decompress(blob)
                 except:
                     logger.error("Failed to read symbol %s" % symbol)
-            return pickle_compat_load(io.BytesIO(data))
+
+            if six.PY2:
+                # Providing encoding is not possible on PY2
+                return pickle_compat_load(io.BytesIO(data))
+            else:
+                try:
+                    return pickle_compat_load(io.BytesIO(data), encoding='bytes')
```

Comment:
How could this ever fail?  Isn't this not decoding?

Reply:
Dicsussed, not the bytes -> str conversion, but the pickled object has other conversions that fail on bytes.  So try the fast option first.  Then fallback to assuming latin which could be unsafe for conversions but makes the other conversions work.